### PR TITLE
PG:clear an unused variable: have

### PR DESCRIPTION
**Pull request by HeyoQiang**
Signed-off-by: Yongqiang He <he.yongqiang@h3c.com>

### CODE REVIEW

**Comment by HeyoQiang** [comment]
@tchaikov How do you think that..?

**Comment by tchaikov** [comment]
lgtm.

**Comment by theanalyst** [comment]
could you squash the commits and use a more descriptive commit message, also commit message missing a singed-off-by

File: src/tools/rbd/action/Snap.cc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+o/ -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
```
Comment (dzafman) [inline src/tools/rbd/action/Snap.cc]: ?????

**Comment by HeyoQiang** [comment]
@dzafman This commit message belong to aonther pull request.  i have reset it.

**Comment by HeyoQiang** [comment]
@dzafman "+o/ -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-"  it is wrong ,i will reset it.

**Comment by dzafman** [comment]
Doesn't build without the variable:

osd/PG.cc: In member function ‘bool PG::choose_acting(pg_shard_t&, bool*)’:
osd/PG.cc:1431:7: error: ‘have’ was not declared in this scope
       have.insert(
       ^
osd/PG.cc:1436:33: error: ‘have’ was not declared in this scope
   if (!(*recoverable_predicate)(have)) {
                                 ^
make[3]: *** [osd/libosd_a-PG.o] Error 1

**Comment by HeyoQiang** [comment]
@dzafman Sorry，i made a mistake. Please look at [https://github.com/ceph/ceph/pull/8713]

**Pull request closed without merging**